apps/admin/src/app/(api)/api/v1/csv/route.ts

```diff
@@ -0,0 +1,97 @@
+import { createHash } from "node:crypto";
+import { env } from "@/env.mjs";
+import { getElectricitySensorIdForUser, getEnergyForSensorInRange } from "@energyleaf/db/query";
+import * as csv from "csv/sync";
+import { type NextRequest, NextResponse } from "next/server";
+
+interface Props {
+    userId: string;
+    userHash: string;
+    start?: string;
+    end?: string;
+}
+
+export async function POST(req: NextRequest) {
+    try {
+        const { userId, userHash, start, end } = (await req.json()) as Props;
+
+        if (!userId || !userHash) {
+            return NextResponse.json(
+                {
+                    error: "Sie haben keinen Zugriff.",
+                },
+                {
+                    status: 401,
+                },
+            );
+        }
+
+        const hash = createHash("sha256").update(`${userId}${env.NEXTAUTH_SECRET}`).digest("hex");
+        if (hash !== userHash) {
+            return NextResponse.json(
+                {
+                    error: "Sie haben keinen Zugriff.",
+                },
+                {
+                    status: 401,
+                },
+            );
+        }
+
+        let sensorId: string | null = null;
+        try {
+            sensorId = await getElectricitySensorIdForUser(userId);
+            if (!sensorId) {
+                throw new Error();
```

Comment:
Wie ist das mit nachträglichen Exports, ist das ein Thema? Ansonsten verlieren wir ja die Export-Möglichkeit nach Testphasen-Ende, wenn der Sensor disconnected wird.
Wobei man nach der Pseudonymisierung ja eh keinen Export mehr gewährleisten kann. Wenn wir die Pseudonymisierung eh sofort nach Testende machen, ist das Thema egal. Sollten wir den Teilnehmern dann vielleicht während des Zeitraums quasi 1 Tag vor Testende oder so nochmal einen Hinweis schicken?

Reply:
Tatsächlich kann er weiterhin seine Daten runterladen.
Die Funktion ist so implementiert, dass nach Abschluss in der History Tabelle geguckt wird, ob dieser Nutzer irgendwann mal einen Sensor hatte.

```typescript
        const history = await trx
            .select()
            .from(sensorHistory)
            .where(and(eq(sensorHistory.userId, userId), eq(sensorHistory.sensorType, SensorType.Electricity)));
```